src/core/tests/frontend/paddle/test_models/gen_scripts/generate_gather_nd.py

```diff
@@ -0,0 +1,52 @@
+#
+# gather_nd paddle model generator
+#
+import numpy as np
+from save_model import saveModel
+import paddle
+import sys
+
+
+def gather_nd(name: str, x, y):
+    paddle.enable_static()
+
+    with paddle.static.program_guard(paddle.static.Program(), paddle.static.Program()):
+        data = paddle.static.data(name="x", shape=x.shape, dtype=x.dtype)
+        index = paddle.static.data(name="index", shape=y.shape, dtype=y.dtype)
+        out = paddle.gather_nd(data, index)
+
+        cpu = paddle.static.cpu_places(1)
+        exe = paddle.static.Executor(cpu[0])
+        # startup program will call initializer to initialize the parameters.
+        exe.run(paddle.static.default_startup_program())
+        outs = exe.run(feed={"x": x, "index": y}, fetch_list=[out])
+
+        saveModel(
+            name,
+            exe,
+            feedkeys=["x", "index"],
+            fetchlist=[out],
+            inputs=[x, y],
+            outputs=[outs[0]],
+            target_dir=sys.argv[1],
+        )
+
+    return outs[0]
+
+
+def main():
+    tests_cases = [
```

Comment:
Looks like we covered enough needed data types test cases here, could you please add some more special cases based on [Paddle gahter_nd unit test](https://github.com/PaddlePaddle/Paddle/blob/release/2.1/python/paddle/fluid/tests/unittests/test_gather_nd_op.py)?  e.g. 'Index has empty element', 'Index has low rank, X has high rank', 'Index has same rank as X's rank', ' Both Index and X have high rank, and Rank(Index) < Rank(X)'. I think each with one test case is enough.

Reply:
Hello, I tested the above case, the following is the result：

## empty index
```python
y = np.array([[], []]).astype("int32")
gather_nd("gather_nd_empty", x, y)
```

![image](https://user-images.githubusercontent.com/27223114/183597636-2af5b298-722d-463e-8a69-2d6a9900c7df.png)

## low rank
```python
x_shape = (10, 20)
x = np.random.rand(*x_shape).astype("float32")
y = np.array([[1], [2]]).astype("int32")
gather_nd("gather_nd_low_index", x, y)
```

![image](https://user-images.githubusercontent.com/27223114/183598380-be22ae60-3bdb-47e6-85d4-4d4b76d4c078.png)

## high rank(float32)
```python
x_shape = (5, 2, 3, 1, 10)
x = np.random.rand(*x_shape).astype("float")
y = np.vstack([np.random.randint(0, s, size=2) for s in x_shape]).T
gather_nd("gather_nd_high_rank1", x, y)

index = np.vstack([np.random.randint(0, s, size=200) for s in x_shape]).T
y = index.reshape([20, 5, 2, 5])
gather_nd("gather_nd_high_rank2", x, y)
```
![image](https://user-images.githubusercontent.com/27223114/183599759-f33e9e30-6aaa-46b3-b644-22eef8702142.png)


So index is empty, high rank (Rank(x)==Rank(index)) should give a hint. But it is very strange that the following code is not passed.

```python
x_shape = (5, 2)
x = np.random.rand(*x_shape).astype("float32")
y = np.vstack([np.random.randint(0, s, size=2) for s in x_shape]).T
gather_nd("gather_nd_high_rank1", x, y)
```

Do you have any other suggestions, thanks for your reviwe.

Reply:
@AndPuQing  :  For 'gather_nd_high_rank1' failed case, could you please have a try the following way?: 
when it failed ,dump the y(index) data, and use this data directly instead of 'np.vstack()' function in 'generate_gather_nd.py' file. e.g.
```
    x_shape = (5, 2, 3, 1, 10)
    x = np.random.rand(*x_shape).astype("float32")
    y = np.array([[0, 0, 1, 0, 4],
                  [4, 0, 0, 0, 1],
                  [0, 1, 2, 0, 2],
                  [0, 1, 2, 0, 6],
                  [1, 1, 2, 0, 3]]).astype("int32")
    gather_nd("gather_nd_high_rank1", x, y)
```
   If this works for our 'gather_nd_high_rank1' test case, that mean there maybe some problem in save or load 'np.vstack()' kind of data in test framework. If so, I think we would better use above index data way instead of 'np.vstack()' for our current cases.

Reply:
I tested the case you mentioned, and it can pass the test. It may indeed be the cause of np.vstack.
![image](https://user-images.githubusercontent.com/27223114/184532678-bdc9854c-5fc9-4238-91b4-497b061fd2cf.png)